File: examples/developer/basic_inbound.ts
/**
 * @file basic_inbound.ts
 * @description Minimal inbound voice agent using Patter SDK with OpenAI Realtime.
 *
 * A friendly restaurant reservation assistant that handles incoming calls.
 * Listens on port 8000 for Twilio webhooks and connects callers to the
 * OpenAI Realtime voice model.
 *
 * Prerequisites:
 *   - OPENAI_API_KEY, TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN in .env
 *   - A Twilio phone number pointed at your webhook URL
 *
 * Usage:
 *   npx ts-node basic_inbound.ts
 */

import { Patter } from "patter";
import dotenv from "dotenv";

dotenv.config();

const phone = new Patter({
  mode: "local",
  openaiKey: process.env.OPENAI_API_KEY!,
  twilioSid: process.env.TWILIO_ACCOUNT_SID!,
  twilioToken: process.env.TWILIO_AUTH_TOKEN!,
  phoneNumber: process.env.TWILIO_PHONE_NUMBER!,
  webhookUrl: process.env.WEBHOOK_URL!,
});

const agent = phone.agent({
  systemPrompt: `You are a friendly reservation assistant for "La Cucina Bella",
an Italian restaurant. Help callers book tables, answer questions about the menu,
and provide directions. Be warm, concise, and professional. If the caller asks
about availability, suggest the next open slot. Always confirm the reservation
details before ending the call.`,
  voice: "nova",
  firstMessage:
    "Hello! Thank you for calling La Cucina Bella. How can I help you today?",
});

async function main(): Promise<void> {
  await phone.serve({ agent, port: 8000 });
  console.log("Inbound agent listening on port 8000");
}

main();

File: examples/developer/basic_inbound.py
"""
Basic Inbound Agent
====================
Minimal inbound voice agent that answers incoming calls as a friendly
restaurant reservation assistant. Uses OpenAI Realtime under the hood --
no custom STT/TTS configuration required.

Requirements:
    pip install patter python-dotenv

Environment variables (.env):
    OPENAI_API_KEY      - OpenAI API key with Realtime access
    TWILIO_ACCOUNT_SID  - Twilio account SID
    TWILIO_AUTH_TOKEN   - Twilio auth token
    TWILIO_PHONE_NUMBER - Your Twilio phone number (E.164 format)
    WEBHOOK_URL         - Public URL where Twilio can reach this server
"""

import asyncio
import os

from dotenv import load_dotenv

load_dotenv()

from patter import Patter

# ── Initialise the Patter client ──────────────────────────────────────
phone = Patter(
    mode="local",
    openai_key=os.getenv("OPENAI_API_KEY"),
    twilio_sid=os.getenv("TWILIO_ACCOUNT_SID"),
    twilio_token=os.getenv("TWILIO_AUTH_TOKEN"),
    phone_number=os.getenv("TWILIO_PHONE_NUMBER"),
    webhook_url=os.getenv("WEBHOOK_URL"),
)

# ── Define the voice agent ────────────────────────────────────────────
agent = phone.agent(
    system_prompt=(
        "You are a friendly reservation assistant for La Dolce Vita, "
        "an Italian restaurant in downtown San Francisco. Help callers "
        "make, modify, or cancel reservations. Ask for the date, time, "
        "party size, and a name for the reservation. Be warm and concise."
    ),
    voice="nova",
    first_message=(
        "Thanks for calling La Dolce Vita! "
        "How can I help you with a reservation today?"
    ),
)

# ── Start the server ──────────────────────────────────────────────────
if __name__ == "__main__":
    asyncio.run(phone.serve(agent, port=8000))
